src/sentry/auth/authenticators/u2f.py

```diff
@@ -116,8 +136,34 @@ def activate(self, request):
 
         return ActivationChallengeResult(challenge=challenge)
 
-    def validate_response(self, request, challenge, response):
+    def validate_response(self, request, challenge, response, is_webauthn_signin_ff_enabled):
         try:
+            if is_webauthn_signin_ff_enabled:
+                # TODO change rp.id later when register is implemented
+                server = U2FFido2Server(
```

Comment:
I don't really know how this library works, but does this have to be defined every function? or can it be a constant?

Reply:
i believe it could be a constant, will make the changes

Reply:
it needs the challenge param that were passed in from validate request so i don't believe it could be taken out of this function

Reply:
what is the `appId`? is that defined as a constant on the frontend?

Reply:
it actually comes from u2f_app_id() in the same class. Im not sure how to access the classproperty while declaring a constant